File: guardias-ies-camp/src/main/java/ies/camp/guardias/repository/entity/Sesion.java
package ies.camp.guardias.repository.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "sesion")
public class Sesion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Profesor profesor;
    private Grupo grupo;
    private Intervalo intervalo;
    private Materia materia;
    private String aula;
}

File: front/guardias-web/src/app/interfaces/sesion.ts
import { Aula } from "./aula";
import { Curso } from "./curso";
import { Dia } from "./dia";
import { Grupo } from "./grupo";
import { Intervalo } from "./intervalo";
import { Materia } from "./materia";
import { Profesor } from "./profesor";

export interface Sesion {
  id: number,
  profesor: Profesor,
  materia: Materia
  grupo: Grupo,
  aula: Aula,
  intervalo: Intervalo,
  curso: Curso,
  dia: Dia,
}
